projects/plugins/jetpack/modules/likes/jetpack-likes-settings.php

```diff
@@ -139,7 +140,7 @@ public function meta_box_save( $post_id ) {
 		// Record sharing disable. Only needs to be done for WPCOM.
 		if ( ! $this->in_jetpack ) {
 			if ( isset( $_POST['post_type'] ) && in_array( $_POST['post_type'], get_post_types( array( 'public' => true ) ), true ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing -- no site changes.
-				if ( ! isset( $_POST['wpl_enable_post_sharing'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing -- no site changes.
+				if ( ! isset( $_POST['wpl_enable_post_sharing'] ) && isset( $_POST['_likesharenonce'] ) && wp_verify_nonce( $_POST['_likesharenonce'], 'likes-and-shares' ) ) { // phpcs:ignore WordPress.Security.ValidatedSanitizedInput -- WordPress core doesn't unslash or verify nonces either.
```

Comment:
This nonce check seems to have been added incorrectly. As written, if `wpl_enable_post_sharing` is set but the nonce check fails, it'll still enable sharing. And the check here doesn't at all protect lines 162, 166, or 176.

I'd have added the nonce check at around line 139 instead. Offhand I don't know if I'd have had it `wp_die()` or just `return $post_id` if the check fails, that depends on whether it'd make more sense to abort or just ignore the attempt.

Then see how many of the phpcs:ignores can be removed.